source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -1599,176 +1599,41 @@ bool SwiftLanguageRuntime::IsValidErrorValue(
 bool SwiftLanguageRuntime::GetDynamicTypeAndAddress_ErrorType(
     ValueObject &in_value, lldb::DynamicValueType use_dynamic,
     TypeAndOrName &class_type_or_name, Address &address) {
-  // layout of error type
-  // pointer to -------> SwiftError {
-  // --------------
-  // CFRuntimeBase
-  // CFIndex
-  // CFStringRef
-  // CFDictionaryRef
-  // --------------
-  // Metadata
-  // WitnessTable
-  // hashable Metadata
-  // hashable WitnessTable
-  // --------------
-  // tail allocated actual object data *
-  // }
-  // * for a struct, it's the inline data
-  // * for a class, it's the inline pointer-to-the-data (aka, the swift class
-  // instance)
-  SwiftErrorDescriptor error_descriptor;
-  if (!IsValidErrorValue(in_value, &error_descriptor))
+  unsigned index = SwiftASTContext::ProtocolInfo::error_instance_index;
+  ValueObjectSP instance_type_sp(
+      in_value.GetStaticValue()->GetChildAtIndex(index, true));
+  if (!instance_type_sp)
     return false;
 
-  Status error;
-  CompilerType var_type(in_value.GetStaticValue()->GetCompilerType());
-  size_t ptr_size = m_process->GetAddressByteSize();
-  SwiftASTContext *swift_ast_ctx =
-      llvm::dyn_cast_or_null<SwiftASTContext>(var_type.GetTypeSystem());
-  if (!swift_ast_ctx)
+  lldb::addr_t existential_address = instance_type_sp->GetValueAsUnsigned(0);
+  if (existential_address == 0 || existential_address == LLDB_INVALID_ADDRESS)
     return false;
 
-  switch (error_descriptor.m_kind) {
-  case SwiftErrorDescriptor::Kind::eNotAnError:
+  // Dynamic type resolution in RemoteAST might pull in other Swift modules, so
+  // use the scratch context where such operations are legal and safe.
+  SwiftASTContext *swift_ast_ctx = GetScratchSwiftASTContext();
```

Comment:
@adrian-prantl Adrian, I saw you changed things here a bit, should this move to your new utility function?

Reply:
the correct new call is probably `in_value.GetScratchSwiftASTContext();`